assets/source/setup-guide/app/components/Account/BusinessAccountSelection.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import { Spinner } from '@woocommerce/components';
+import { addQueryArgs } from '@wordpress/url';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	CardBody,
+	Flex,
+	FlexBlock,
+	FlexItem,
+	SelectControl,
+	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
+} from '@wordpress/components';
+
+const BusinessAccountSelection = ( {
+	businessAccounts,
+	setAttemptedCreation,
+} ) => {
+	const [ targetBusinessId, setTargetBusinessId ] = useState(
+		undefined !== businessAccounts && businessAccounts.length > 0
+			? businessAccounts[ 0 ][ 'value' ]
+			: null
+	);
+
+	const handleConnectToBusiness = () => {
+		const newURL = addQueryArgs(
+			decodeEntities(
+				wcSettings.pinterest_for_woocommerce.switchBusinessAccountUrl
+			),
```

Comment:
The `switchBusinessAccountUrl` is a piece of data, we don't need to encode it in server side and decode it in JS here. If we do want to display it in HTML, we can encode it in JS.

Reply:
It's `esc_url()` in the backend that encodes that. Imho its better to have that and decode it in the frontend. 
I can remove though if its critical.

Reply:
Yeah I noticed the `esc_url()`.

> Imho its better to have that and decode it in the frontend.

But, why? 🤔  This is similar to PR https://github.com/woocommerce/pinterest-for-woocommerce/pull/118#discussion_r682069238. 

It's not at critical level, but I just don't see the need for it.